src/components/NewBookForm.js

```diff
@@ -1,22 +1,24 @@
 import React, { useState } from 'react';
 import { useDispatch } from 'react-redux';
-import { addBook } from '../redux/books/BookSlice';
+import { addBookAsync } from '../redux/books/BookSlice';
 
 const NewBookForm = () => {
   const [title, setTitle] = useState('');
   const [author, setAuthor] = useState('');
+  const [category, setCategory] = useState('Fiction');
   const dispatch = useDispatch();
 
-  const handleSubmit = (e) => {
+  const handleSubmit = async (e) => {
     e.preventDefault();
     const newBook = {
-      id: Math.random().toString(),
       title,
       author,
+      category,
     };
-    dispatch(addBook(newBook));
+    await dispatch(addBookAsync(newBook));
     setTitle('');
     setAuthor('');
+    setCategory('Fiction');
   };
```

Comment:
Currently, when adding a book, the book added doesn't contain the title and author specified in form. Instead it displays blank text as seen in the video below:


https://github.com/mpa-mxiang/bookstores/assets/37116322/d7362448-26e8-4485-96d2-9aeff315bb65

Please update your application so that books are added without bugs🙏